### PR TITLE
Keep manually selected language

### PR DESCRIPTION
**Pull request by bekicot**
Keep manually selected language
fix: #60 
On Chrome:
fix 60 chrome
On Firefox:
fix 60 firefox

@danielnaber  It also introduce a new api for tool and a new class called callback. Which we can definitely use for something else. Please let me know though.

### CODE REVIEW

File: webextension/callback.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    if(!this.callbacks) {
+      this.callbacks = {};
+    }
+    return this.callbacks;
+  }
+}
+
+if (typeof module !== 'undefined') {
+    module.exports = Callback;
+}
```
Comment (jayvdb) [inline webextension/callback.js]: no EOL at EOF

File: webextension/callback.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class Callback {
+  constructor() {}
```
Comment (jayvdb) [inline webextension/callback.js]: this repo uses four space indents.

File: webextension/callback.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  static add(name, callback) {
+    // initialize callback;
+    this.getCallbacks();
+    if(this.callbacks[name] === undefined) {
```
Comment (jayvdb) [inline webextension/callback.js]: `if (`

File: webextension/callback.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  static run(name) {
+    var callbacks = this.getCallbacks();
+    for(let callback of callbacks[name]) {
+        callback.call();
+    }
+    return true;
+  }
+
+  static getCallbacks() {
+    if(!this.callbacks) {
```
Comment (jayvdb) [inline webextension/callback.js]: should be `if (`

File: webextension/callback.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    if(this.callbacks[name] === undefined) {
+      this.callbacks[name] = [callback];
+    } else {
+      this.callbacks[name].push(callback);
+    }
+    return true;
+  }
+  static run(name) {
+    var callbacks = this.getCallbacks();
+    for(let callback of callbacks[name]) {
```
Comment (jayvdb) [inline webextension/callback.js]: `for (`

File: webextension/popup.js [hunk 6]
```diff
@@ -550,3 +551,41 @@ document.addEventListener('DOMContentLoaded', function() {
...
     return chrome.storage.sync ? chrome.storage.sync : chrome.storage.local;
 }
+
+function languageSelection() {
+    return document.getElementById('language');
+}
+
+function saveLanguageSelectionToStorage() {
+    let storage            = getStorage();
+    if(languageSelection()) {
```
Comment (jayvdb) [inline webextension/popup.js]: `if (`

File: webextension/popup.js [hunk 7]
```diff
@@ -550,3 +551,41 @@ document.addEventListener('DOMContentLoaded', function() {
...
+        languageSelection().addEventListener('change', (event) => {
+            storage.get('siteLanguage', (result) => {
+                // getting current hostname and path
+                Tools.currentTab((tab) => {
+                    let listOfLanguages = result.siteLanguage || {};
+                    listOfLanguages[Tools.hostAndPath(tab.url)] = event.target.value;
+                    // listOfLanguages[currentHostAndPath()] =
```
Comment (jayvdb) [inline webextension/popup.js]: is this needed?

File: webextension/popup.js [hunk 8]
```diff
@@ -550,3 +551,41 @@ document.addEventListener('DOMContentLoaded', function() {
...
+                    });
+                });
+            });
+        });
+    }
+}
+
+function setLanguageSelectionValue() {
+    let languageElm = languageSelection();
+    if(languageElm) {
```
Comment (jayvdb) [inline webextension/popup.js]: `if (`

**Comment by bekicot** [comment]
@jayvdb Cool 😄 . 
All of the suggestion have been applied 👍 .

**Comment by danielnaber** [comment]
Closing old issues, as this repo is not used anymore

**Pull request closed without merging**